### PR TITLE
New gallery performance improvements

### PR DESCRIPTION
**Pull request by xster**
Use https://github.com/flutter/flutter/pull/17167/files?&w=1 to ignore whitespaces.

### CODE REVIEW

File: examples/flutter_gallery/lib/gallery/home.dart [hunk 1]
```diff
@@ -52,37 +52,40 @@ class _CategoryItem extends StatelessWidget {
...
-          new Container(
-            height: 48.0,
-            alignment: Alignment.center,
-            child: new Text(
-              category.name,
-              textAlign: TextAlign.center,
-              style: theme.textTheme.subhead.copyWith(
-                fontFamily: 'GoogleSans',
+    // Contain the repaint from the splash.
+    return new RepaintBoundary(
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/gallery/home.dart]:
```
// This repaint boundary prevents the entire app from being rebuilt when 
// the _CategoriesPage is rebuilt due to the backdrop's front layer opening
// or closing.
```

**Review by HansMuller — APPROVED** [review]
For the record, please explain how this performance improvement was measured (how it will be monitored going forward), how much of an improvement this change delivers.

I'm hopeful: LGTM

**Comment by xster** [comment]
Sure. Debugged with either `debugRepaintRainbowEnabled` or with `debugProfileBuildsEnabled` and `debugProfileRepaintsEnabled` and looking at the timeline. 

Fixes redundant repainting of static render object subtrees during the crossfade animation and during the show options toggle. 

Should improve average_frame_build_time_millis for flutter_gallery_ios__transition_perf and flutter_gallery__transition_perf

File: examples/flutter_gallery/lib/gallery/home.dart [hunk 2]
```diff
@@ -52,37 +52,40 @@ class _CategoryItem extends StatelessWidget {
...
-          const SizedBox(height: 10.0),
-          new Container(
-            height: 48.0,
-            alignment: Alignment.center,
-            child: new Text(
-              category.name,
-              textAlign: TextAlign.center,
-              style: theme.textTheme.subhead.copyWith(
-                fontFamily: 'GoogleSans',
+    // Contain the repaint from the splash.
```
[the remark this answers is not in the document]
Comment (xster) [reply examples/flutter_gallery/lib/gallery/home.dart]: Took your wording

File: examples/flutter_gallery/lib/gallery/home.dart (hunk 1 above)
Comment (xster) [reply examples/flutter_gallery/lib/gallery/home.dart]: It's the other way around. Reworded a bit.

**Pull request merged**